### PR TITLE
Add egress kong for authentic id proxy

### PR DESCRIPTION
**Pull request by dzaslavskiy**
Adding a kong egress instance.

The first use case for it is to proxy authenticid

### CODE REVIEW

File: egress/kong-config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        paths:
+          - /authenticid
+        methods:
+          - GET
+          - POST
+          - PUT
+          - DELETE
+          - OPTIONS
+        hosts:
+          - idva-kong-egress-${ENVIRONMENT_NAME}.apps.internal
```
Comment (folksgl) [inline egress/kong-config.yaml:11]: What is the effect of having both a `paths` and a `hosts` section? Does both `/` and `/authenticid` now direct to the upstream?

File: egress/kong-config.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        protocols:
+          - http
```
Comment (folksgl) [inline egress/kong-config.yaml:13]: Should this not be `https`? If not, why?

File: kong-config.yaml [hunk 3]
```diff
@@ -40,5 +40,22 @@ services:
...
+        methods:
+          - GET
+          - POST
+          - PUT
+          - DELETE
+          - OPTIONS
+        hosts:
+          - idva-${ENVIRONMENT_NAME}.app.cloud.gov
+        protocols:
+          - http
```
Comment (folksgl) [inline kong-config.yaml:53]: Should this not be https? If not, why?

File: egress/kong-config.yaml (hunk 1 above)
Comment (dzaslavskiy) [reply egress/kong-config.yaml:11]: The conditions both must be true to match the route. only requests to /authenticid with host idva-kong-egress-${ENVIRONMENT_NAME}.apps.internal

File: egress/kong-config.yaml (hunk 2 above)
Comment (dzaslavskiy) [reply egress/kong-config.yaml:13]: kong is behind a tls terminating proxy so it should be http

**Review by folksgl — APPROVED** [review]

File: kong-config.yaml [hunk 4]
```diff
@@ -40,5 +40,16 @@ services:
           add:
             headers:
             - X-Frame-Options:ALLOWALL
+  - name: authenticid
+    url: https://idva-kong-egress-${ENVIRONMENT_NAME}.apps.internal:61443
```
Comment (folksgl) [inline kong-config.yaml:44]: Ahh don't we need `/authenticid` as the path here?
```suggestion
    url: https://idva-kong-egress-${ENVIRONMENT_NAME}.apps.internal:61443/authenticid
```

Comment (dzaslavskiy) [reply kong-config.yaml:44]: strip_path: false keeps the path component on the request

**Review by folksgl — APPROVED** [review]

**Pull request merged**